firmware/src/imu/mod.rs

```diff
@@ -12,13 +12,44 @@ use crate::{
 };
 
 pub type Quat = nalgebra::UnitQuaternion<f32>;
+pub type Accel = nalgebra::Vector3<f32>;
+pub type Gyro = nalgebra::Vector3<f32>;
 
-pub trait FusedImu {
-	type Error: core::fmt::Debug;
+pub struct UnfusedData {
+	accel: Accel,
+	gyro: Gyro,
+}
+
+pub struct FusedData {
+	quat: Quat,
+}
+
+/// Represents a sensor fusion algorithm that will take an imu's `UnfusedData` and do math to turn
+/// it into `FusedData`, suitable for use as orientation.
+pub trait Fuser {
+	// TODO: Does a one-in, one-out api here work? Should we reuse a standard trait like a
+	// sink/stream/iterator?
+	// Note: Intentionally not async rn, this should only be doing math, not io or any internal
+	// awaiting.
+	fn process(&mut self, unfused: &UnfusedData) -> FusedData;
```

Comment:
You might be interested in making this a generic trait (or one with an associated type) to take in multiple kinds of unfused data.

quick sketch:
```rust
pub trait Fuser<UnfusedData> {
    fn process(&mut self, unfused: &UnfusedData) -> FusedData;
}

impl Fuser<(Gyro, Accel)> for VQF { /* ... */ }
impl Fuser<(Gyro, Accel, Mag)> for VQF { /* ... */ }

// this might need a where clause?
pub struct FusedImu<I: Imu, F: Fuser<<I as Imu>::Data>> { /* ... */ }
```

this would allow for e.g. a data type that has no magnetometer readings, and a data type that has magnetometer readings.

Reply:
Agreed - I actually broke the fuser out as a separate trait and have separate wrapper structs for fused and unfused is in part because that refactor may be necessary later.
For now, I'm not interested in 9dof so any genericism on *what* the fuser is processing is unecessary though.